internal/executor/util/process.go

```diff
@@ -0,0 +1,37 @@
+package util
+
+import (
+	"context"
+	"sync"
+
+	commonUtil "github.com/armadaproject/armada/internal/common/util"
+)
+
+func ProcessItemsWithThreadPool[K any](ctx context.Context, maxThreadCount int, itemsToProcess []K, processFunc func(K)) {
+	wg := &sync.WaitGroup{}
+	processChannel := make(chan K)
+
+	for i := 0; i < commonUtil.Min(len(itemsToProcess), maxThreadCount); i++ {
```

Comment:
pretty sure there's a `min` function in the standard library now

Reply:
Are you sure? I can't seem to find it - would you mind linking to anywhere we use it?

New go does allow writing a generic min function - so you're probably correct

Reply:
ah I'm wrong- it doesn't yet exist.  Hopefully it will soon